StRoot/StGeant4Maker/StGeant4Maker.cxx

```diff
@@ -0,0 +1,1539 @@
+#include "StGeant4Maker.h"
+//________________________________________________________________________________________________
+#include "StMessMgr.h"
+//________________________________________________________________________________________________
+#include "StarMagField.h"
+#include "StMCParticleStack.h"
+#include "TSystem.h"
+#include "StBFChain.h"
+#include "TInterpreter.h"
+#include "TGeoManager.h"
+#include "StarVMC/StarAgmlLib/StarAgmlStacker.h"
+#include "StarGenerator/BASE/StarPrimaryMaker.h"
+#include "StarGenerator/BASE/StarParticleStack.h"
+#include "StarGenerator/UTIL/StarParticleData.h"
+#include "StarGenerator/UTIL/StarRandom.h"
+
+#include "TString.h"
+#include "StSensitiveDetector.h"
+
+#include <CLHEP/Random/Random.h>
+
+#include "StarVMC/StarAgmlLib/AgMLExtension.h"
+#include "GeometryUtils.h"
+#include "TString.h"
+
+#include "StChain/StEvtHddr.h"
+#include "TH2F.h"
+
+#include <functional>
+
+//_______________________________________________________________________________________________
+#include <AgMLVolumeIdFactory.h>
+//_______________________________________________________________________________________________
+#include "TMCManager.h"
+//________________________________________________________________________________________________
+#include "TGeant4.h"
+#include "TG4RunManager.h"
+#include "TG4RunConfiguration.h"
+#include "TGeant3TGeo.h"
+//________________________________________________________________________________________________
+#include "tables/St_g2t_event_Table.h"
+#include "tables/St_g2t_vertex_Table.h"
+#include "tables/St_g2t_track_Table.h"
+//________________________________________________________________________________________________
+#include "g2t/St_g2t_tpc_Module.h"
+#include "g2t/St_g2t_hca_Module.h"
+#include "g2t/St_g2t_wca_Module.h"
+#include "g2t/St_g2t_pre_Module.h"
+#include "g2t/St_g2t_fts_Module.h"
+#include "g2t/St_g2t_stg_Module.h"
+#include "g2t/St_g2t_epd_Module.h"
+#include "g2t/St_g2t_tfr_Module.h"
+#include "g2t/St_g2t_mtd_Module.h"
+#include "g2t/St_g2t_vpd_Module.h"
+//________________________________________________________________________________________________
+#include <StHitCollection.h> 
+#include <cstring>
+//________________________________________________________________________________________________
+
+// Functors used to copy the hits from the sensitive detector hit collections into the g2t tables.
```

Comment:
These functors are the equivalent of the `pams/sim/g2t/g2t_???.F` functions, which copy the zebra hits into the c++ tables.